### PR TITLE
VMware: Refactor vmware_vswitch

### LINKED ISSUE #36030: Refactor vmware_vswitch is required
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

```
vmware_vswitch
```
##### ANSIBLE VERSION

```
2.5.0a1
```

##### SUMMARY

```
when i have switch created on certain nic and trying to create another switch with same name on another nic i should get error but ansible module didnt return error instead returned success
```
##### STEPS TO REPRODUCE

```
---
- hosts: 127.0.0.1
  connection: local
  tasks:
    - name: Add a VMware vSwitch
      action:
        module: vmware_vswitch
        hostname: 10.243.1.43
        username: root
        password: PASSW0RD
        switch: vSwitch0
        nics: vmnic0
        mtu: 1500
        validate_certs: no
        state: present
---
- hosts: 127.0.0.1
  connection: local
  tasks:
    - name: Add a VMware vSwitch
      action:
        module: vmware_vswitch
        hostname: 10.243.1.43
        username: root
        password: PASSW0RD
        switch: vSwitch0
        nics: vmnic1
        mtu: 1500
        validate_certs: no
        state: present


```

##### EXPECTED RESULTS

```
expect error as switch is already attached to vmnic0 or at end of script switch should get attached to both nics
```
##### ACTUAL RESULTS

```
 playbook executed successfully and Vswitch0 is still attached to only vmnic0
```

```

```

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
* Update documentation
* Update logic
* Added idempotency
* Added Error handling

Fixes: #36030

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/vmware/vmware_vswitch.py

##### ANSIBLE VERSION

```
2.6devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@pdellaert @tchernomax Could you please review this ?

File: lib/ansible/modules/cloud/vmware/vmware_vswitch.py [hunk 1]
```diff
@@ -133,96 +140,260 @@ def __init__(self, module):
         self.nics = module.params['nics']
         self.mtu = module.params['mtu']
         self.state = module.params['state']
-        self.esxi_hostname = module.params['esxi_hostname']
+        esxi_hostname = module.params['esxi_hostname']
+
+        hosts = get_all_objs(self.content, [vim.HostSystem])
```
Comment (pdellaert) [inline lib/ansible/modules/cloud/vmware/vmware_vswitch.py]: Can we use the refactored logic of host/cluster here to simplify?

Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_vswitch.py]: Yes.

**Review by tchernomax — APPROVED** [review]
LGTM

**Comment by umrikar** [comment]
@Akasurde could you tell when this change will get merged

**Comment by Akasurde** [comment]
@tchernomax @pdellaert @umrikar Thanks for the reviews.

**Pull request merged**